src/procedures/__tests__/CreateCheckpoint.ts

```diff
@@ -50,8 +74,71 @@ describe('CreateCheckpoint', () => {
 
   describe('createCheckpoint', () => {
     test('should send the transaction to createCheckpoint', async () => {
+      const addTransactionSpy = spy(target, 'addTransaction');
+
       // Real call
       await target.prepareTransactions();
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(securityTokenMock.getMockInstance().controllerTransfer)
+      ).toEqual(true);
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should throw if corresponding checkpoint event is not fired', async () => {
+      findEventsStub = ImportMock.mockFunction(utilsModule, 'findEvents', []);
+
+      // Real call
+      const resolver = await target.prepareTransactions();
+
+      expect(resolver.run({} as TransactionReceiptWithDecodedLogs)).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.UnexpectedEventLogs,
+          message:
+            "The Checkpoint was successfully created but the corresponding event wasn't fired. Please report this issue to the Polymath team.",
+        })
+      );
+    });
+
+    test('should correctly return the resolver', async () => {
+      const checkpointObject = {
+        checkpoint: {
+          securityTokenId: () => Promise.resolve(params1.symbol),
+          index: () => Promise.resolve(1),
+        },
+      };
+      const fetchStub = checkpointFactoryMock.mock('fetch', checkpointObject);
```

Comment:
Strictly speaking, all `fetch` functions return their values wrapped in promises